pkg/pool-cow/test/foundry/CowRouter.t.sol

```diff
@@ -604,6 +604,110 @@ contract CowRouterTest is BaseCowTest {
         assertEq(cowRouter.getProtocolFeePercentage(), newProtocolFeePercentage, "Protocol Fee Percentage is not set");
     }
 
+    /********************************************************
+                          getFeeSweeper()
+    ********************************************************/
+    function testGetFeeSweeper() public {
+        assertEq(cowRouter.getFeeSweeper(), feeSweeper, "Wrong fee sweeper");
+
+        address newFeeSweeper = address(1);
+
+        vm.prank(admin);
+        cowRouter.setFeeSweeper(newFeeSweeper);
+
+        assertEq(cowRouter.getFeeSweeper(), newFeeSweeper, "Fee sweeper was not set properly");
+    }
+
+    /********************************************************
+                          setFeeSweeper()
+    ********************************************************/
+    function testSetFeeSweeperIsPermissioned() public {
+        vm.expectRevert(IAuthentication.SenderNotAllowed.selector);
+        cowRouter.setFeeSweeper(address(1));
+    }
+
+    function testSetFeeSweeperInvalidAddress() public {
+        // Address 0 is not a valid fee sweeper.
+        vm.expectRevert(abi.encodeWithSelector(ICowRouter.InvalidFeeSweeper.selector));
+        vm.prank(admin);
+        cowRouter.setFeeSweeper(address(0));
+    }
+
+    function testSetFeeSweeper() public {
+        address newFeeSweeper = address(2);
+
+        vm.prank(admin);
+        vm.expectEmit();
+        emit ICowRouter.FeeSweeperChanged(newFeeSweeper);
+        cowRouter.setFeeSweeper(newFeeSweeper);
+
+        assertEq(cowRouter.getFeeSweeper(), newFeeSweeper, "Fee sweeper was set properly");
+    }
+
+    /********************************************************
```

Comment:
Probably don't need a section header here, as there's only one function (unless you're going to add more later?). Actually, it would be a good idea to keep the section header (something like "Withdraw Protocol Fees", and separate this one long test out into a few individual tests (e.g., event, token with zero balance, if you send the router "excess" tokens it still only withdraws the calculated amount, etc.).